### DISCUSSION TITLE
en_core_web_trf  - taking 10 secs to process text with 1000 words

### DISCUSSION THREAD

**Discussion by mattkallo** [explosion/spaCy#11850] [Help: Model Advice] [answered]
Is it normal for en_core_web_trf to take more than 10 secs to process? I have seen other discussions regarding speed, and have tried disabling components and setting thread count as suggested, but still the prediction seems to be taking more than 10 secs.  Could anything else degrade performance/speed? What should I be checking? Thanks

Input text is english news article with :  1,095 words (6,810 characters)

Code:

```
        torch.set_num_threads(1)
        nlp = spacy.load("en_core_web_trf")
        nlp.disable_pipe("ner")
        nlp.disable_pipe("parser")
        doc = nlp(text)

```

spaCy version: 3.0.2
Model: en_core_web_trf

CPU : 
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              32
On-line CPU(s) list: 0-31
Thread(s) per core:  2
Core(s) per socket:  16

RAM:  128GB

Environment : Docker container with Ubuntu
Python : 3.7.5
No GPU

**Comment by polm** [comment, marked as answer]
The transformer components are designed to be used on a GPU. While you can use them without a GPU - and that's important from the perspective of code correctness - they will be extremely slow, so it's not recommended.
